### DISCUSSION TITLE
ModuleNotFoundError: No module named 'tufup'

### DISCUSSION THREAD

**Discussion by NguyenDucQuan12** [dennisvang/notsotuf-example#34] [Q&A]
Hello, when I try to run your example, I get an error that module name 'tufup' cannot be installed.
![image](https://github.com/dennisvang/tufup-example/assets/68120446/c19735fc-b393-4fba-8d6e-f76147c314fb)
![image](https://github.com/dennisvang/tufup-example/assets/68120446/5c493b58-61d1-418d-ba85-c53ea355af66)

**Comment by NguyenDucQuan12** [comment]
first try `python -m pip install --upgrade --force-reinstall pip` and then do `python -m pip freeze`
for those who encounter errors like me when using virtual environments

**Comment by NguyenDucQuan12** [comment]
![image](https://github.com/dennisvang/tufup-example/assets/68120446/c7b5b012-6953-4c86-9953-eea821d84c4c)
I have installed the required files, but I still get this error. Please tell me how to fix it
```
python -m pip freeze
altgraph==0.17.4
bsdiff4==1.2.4
certifi==2024.2.2
cffi==1.16.0
charset-normalizer==3.3.2
cryptography==42.0.5
idna==3.6
packaging==24.0
pefile==2023.2.7
pycparser==2.22
pyinstaller==6.5.0
pyinstaller-hooks-contrib==2024.3
PyNaCl==1.5.0
pywin32-ctypes==0.2.2
requests==2.31.0
ruff==0.3.5
securesystemslib==0.31.0
tuf==3.1.1
tufup @ git+https://github.com/dennisvang/tufup.git@9e53b744a8811d0d9ab6820172da11c9a4327453
urllib3==2.2.1
```

**Comment by dennisvang** [comment]
@NguyenDucQuan12

What happens if you run `tufup -v` on the command line?

**Comment by NguyenDucQuan12** [comment]
@dennisvang  return  `tufup 0.7.1`
![image](https://github.com/dennisvang/tufup-example/assets/68120446/dc6a46ff-c697-4d93-a899-c544e0cb5018)
![image](https://github.com/dennisvang/tufup-example/assets/68120446/2f630f79-fea0-46fc-b540-b5104d06656b)

**Comment by dennisvang** [comment]
@NguyenDucQuan12 This looks like an IDE configuration issue: 

tufup appears to be installed properly in your `venv`, but your IDE (VS Code?) does not appear to be using that same `venv`.

I'm not familiar with VS Code/Pylance, but perhaps it's related to [this issue][1]?

[1]: https://github.com/microsoft/pylance-release/issues/330
